From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Fri, 15 Jan 2016 13:57:47 +0100
Subject: [PATCH] {NOT UPSTREAMABLE} net/core/sock_reuseport: I did not
 understand this clang error

  CC      net/core/sock_reuseport.o
net/core/sock_reuseport.c:103:2: error: ignoring return value of function declared with const attribute [-Werror,-Wunused-value]
        WARN_ONCE(rcu_dereference_protected(sk->sk_reuseport_cb,
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
include/asm-generic/bug.h:126:2: note: expanded from macro 'WARN_ONCE'
        unlikely(__ret_warn_once);                              \
        ^~~~~~~~~~~~~~~~~~~~~~~~~
include/linux/compiler.h:166:22: note: expanded from macro 'unlikely'
                        ^~~~~~~~~~~~~~~~ ~~~~~~~~
1 error generated.
---
 net/core/sock_reuseport.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/core/sock_reuseport.c b/net/core/sock_reuseport.c
index b1e0dbea1e8c..5039b45518a0 100644
--- a/net/core/sock_reuseport.c
+++ b/net/core/sock_reuseport.c
@@ -112,7 +112,7 @@ int reuseport_add_sock(struct sock *sk, struct sock *sk2)
 
 	spin_lock_bh(&reuseport_lock);
 	reuse = rcu_dereference_protected(sk2->sk_reuseport_cb,
-					  lockdep_is_held(&reuseport_lock)),
+					  lockdep_is_held(&reuseport_lock));
 	WARN_ONCE(rcu_dereference_protected(sk->sk_reuseport_cb,
 					    lockdep_is_held(&reuseport_lock)),
 		  "socket already in reuseport group");
-- 
